text/src/swam/witx/parser/TypesParser.scala

```diff
@@ -0,0 +1,102 @@
+package swam
+package witx
+package parser
+
+import fastparse._
+import swam.witx.parser.parser.WitxWhitespace._
+import swam.witx.unresolved._
+
+import scala.collection.MapView
+
+/**
+    @author Javier Cabrera-Arteaga on 2020-03-18
+  */
+object TypesParser {
+
+  import Lexical._
+
+  private var declaredTypes = Map[String, BaseWitxType](
+    "u32" -> BasicType("u32"),
+    "u64" -> BasicType("u64"),
+    "u8" -> BasicType("u8"),
+    "u16" -> BasicType("u16"),
+    "string" -> BasicType("string")
+  )
+
+  def file[_: P]: P[Map[String, BaseWitxType]] = {
+    P(ws ~ types ~ ws ~ End).map(_ => declaredTypes)
```

Comment:
Why always return `declaredTypes` and discard the actually declared ones?

Reply:
I am not sure if I understand correctly. The ```declaredTypes``` instance is being updated after every type parsing rule. I think I am not discarding anything.